### PR TITLE
Add missing MetricFrame tests

### PR DESCRIPTION
**Pull request by riedgar-ms**
The `MetricFrame` smoke tests didn't cover cases with `sample_params` (there are tests of `sample_params` but they check the internal methods, and not the API which users would invoke). Add a couple of tests to rectify this oversight.

### CODE REVIEW

**Review by romanlutz — APPROVED** [review]
Thanks for adding these!!!

File: test/unit/metrics/test_metricframe_smoke.py [hunk 1]
```diff
@@ -226,6 +226,95 @@ def test_1m_1sf_1cf_metric_dict(transform_y_t, transform_y_p):
     assert target_maxs['recall']['m'] == max(recall_m_arr)
 
 
+def test_1m_1_sf_sample_weights():
+
+    def multi_sp(y_t, y_p, p1, p2):
+        assert len(y_t) == len(y_p)
```
Comment (adrinjalali) [inline test/unit/metrics/test_metricframe_smoke.py:232]: could you please add a comment about what this function does?

File: test/unit/metrics/test_metricframe_smoke.py [hunk 2]
```diff
@@ -226,6 +226,95 @@ def test_1m_1sf_1cf_metric_dict(transform_y_t, transform_y_p):
     assert target_maxs['recall']['m'] == max(recall_m_arr)
 
 
+def test_1m_1_sf_sample_weights():
```
Comment (adrinjalali) [inline test/unit/metrics/test_metricframe_smoke.py:229]: a small comment here wouldn't hurt ;)

File: test/unit/metrics/test_metricframe_smoke.py [hunk 3]
```diff
@@ -226,6 +226,95 @@ def test_1m_1sf_1cf_metric_dict(transform_y_t, transform_y_p):
...
+                                 sensitive_features=g_1,
+                                 sample_params={'p1': param1, 'p2': param2})
+
+    # Sanity check types
+    assert isinstance(target.overall, float)
+    assert isinstance(target.by_group, pd.Series)
+    assert target.by_group.shape == (2,)
+
+    assert target.overall == sum(param2)
+    for g in g_1:
```
Comment (adrinjalali) [inline test/unit/metrics/test_metricframe_smoke.py:253]: what's `g_1`? It's imported up there, but it's really not easy to read this code and understand what it does.

**Review by adrinjalali — COMMENTED** [review]
Thanks @riedgar-ms , same comments apply for the second test :)

File: test/unit/metrics/test_metricframe_smoke.py (hunk 2 above)
Comment (riedgar-ms) [reply test/unit/metrics/test_metricframe_smoke.py:230]: Done

File: test/unit/metrics/test_metricframe_smoke.py (hunk 3 above)
Comment (riedgar-ms) [reply test/unit/metrics/test_metricframe_smoke.py:266]: Added a note here, and up at the top where it's imported

File: test/unit/metrics/test_metricframe_smoke.py (hunk 1 above)
Comment (riedgar-ms) [reply test/unit/metrics/test_metricframe_smoke.py:238]: Done

**Comment by riedgar-ms** [comment]
While there were tests of `sample_params` there were not in the end-to-end flows which users would invoke, but instead called internals.

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**